src/app/recommendations.ts

```diff
@@ -296,4 +296,187 @@ export const RECOMMENDATIONS: Step[] = [
   { possibleIn: 1700, necessaryAsOf: 1700, level: ApplicationComplexity.Medium, step: 'v17 withNoDomReuse', action: 'To disable hydration use `ngSkipHydration` or remove the `provideClientHydration` call from the provider list since `withNoDomReuse` is no longer part of the public API.' },
   { possibleIn: 1700, necessaryAsOf: 1700, level: ApplicationComplexity.Basic, step: 'v17 paramsInheritanceStrategy', action: 'If you want the child routes of `loadComponent` routes to inherit data from their parent specify the `paramsInheritanceStrategy` to `always`, which in v17 is now set to `emptyOnly`.' },
 
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Advanced,
+    step: "18.0.0: async has been removed, use waitForAsync instead",
+    action: "Replace `async` from `@angular/core` with `waitForAsync`."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Advanced,
+    step: "18.0.0: Deprecated matchesElement method removed from AnimationDriver",
+    action: "Remove calls to `matchesElement` because it's now not part of `AnimationDriver`."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Medium,
+    step: "18.0.0. Use `@angular/core` StateKey and TransferState",
+    action: "Import `StateKey` and `TransferState` from `@angular/core` instead of `@angular/platform-browser`."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Medium,
+    step: "18.0.0. Opt-out of caching for HTTP requests with auth headers",
+    action: "Use `includeRequestsWithAuthHeaders: true` in `withHttpTransferCache` to opt-out of caching for HTTP requests that require authorization."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Advanced,
+    step: "18.0.0.REMOVE_OBSOLETE_IS_WORKER",
+    action: "Update the application to remove 'isPlatformWorkerUi' and 'isPlatformWorkerApp' since they were part of platform WebWorker which is now not part of Angular."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Medium,
+    step: "18.0.0.FORCE_ZONE_CHANGE_DETECTION",
+    action: "Your tests may run additional rounds of change detection. Revert to the old behavior by adding 'provideZoneChangeDetection({schedulingMode: NgZoneSchedulingMode.NgZoneOnly})' to the TestBed providers."
+  },
+  {
+    possibleIn: 1800,
+    necessaryAsOf: 1800,
+    level: ApplicationComplexity.Medium,
+    step: "18.0.0.DISABLE_ZONE_CHANGE_DETECTION",
+    action: "If you observe performance regressions due to extra change detection cycles, update the application to add 'provideZoneChangeDetection({schedulingMode: NgZoneSchedulingMode.NgZoneOnly})' to the providers in 'bootstrapApplication' or 'BootstrapOptions' of 'bootstrapModule' to prevent change detection when updating state outside of the zone."
```

Comment:
```suggestion
```

This one is the same as the last one, but for applications. We haven't yet encountered applications that _should_ disable this behavior.